![Illustration of silhouetted heads](mentalhealth.jpg)

Does going to university in a different country affect your mental health? A Japanese international university surveyed its students in 2018 and published a study the following year that was approved by several ethical and regulatory boards.

The study found that international students have a higher risk of mental health difficulties than the general population. Explore the `students` data using PostgreSQL to find out if this is true and see if the length of stay is a contributing factor.

Here is a data description of the fields you may find helpful. The full dataset is in one table with 50 fields and, according to the survey, 268 records. Each row is a student.

| Field Name    | Description                                      | 
| ------------- | ------------------------------------------------ |
| inter_dom     | Types of students                                |
| japanese_cate | Japanese language proficiency                    | 
| english_cate  | English language proficiency                     |
| academic      | Current academic level                           | 
| age           | Current age of student                           |
| stay          | Current length of stay in years                  |
| todep         | Total score of depression (PHQ-9 test)           |
| tosc          | Total score of social connectedness (SCS test)   |
| toas          | Total score of Acculturative Stress (ASISS test) |

Your task will be to do the following exploratory analysis:

- Count the number of all records, and all records per student type
- Filter the data to see how it differs between the student types
- Find the summary statistics of the diagnostic tests for all students
- Summarize the data for international students
- See if length of stay impacts the test scores

In [7]:
SELECT * FROM students LIMIT 5;

inter_dom region  gender academic  ...  religion_bi  alone_bi  others_bi internet_bi
0     Inter    SEA    Male     Grad  ...           No        No         No          No
1     Inter    SEA    Male     Grad  ...           No        No         No          No
2     Inter    SEA    Male     Grad  ...           No        No         No          No
3     Inter     EA  Female     Grad  ...           No        No         No          No
4     Inter     EA  Female     Grad  ...          Yes        No         No          No

[5 rows x 50 columns]

Task 1: Count the number of all record, and all record per student type
----------------------------------------------------------------------------
**Start by exploring the dataset and understanding the data you are working with**
Count the number of records in the dataset to confirm you have the expected number of records, then see how many records you have for each student type. Use the aliases total_records and count_inter_dom.

In [1]:
-- Start coding here...
SELECT 
COUNT(*) AS total_records 
FROM students;


total_records
0            286

In [3]:
SELECT 
inter_dom,
COUNT(inter_dom) AS count_inter_dom
FROM students
GROUP BY inter_dom
ORDER BY inter_dom;


inter_dom  count_inter_dom
0                         18
1       Dom               67
2     Inter              201

Task 2: Create a filter to explore how the data is different in each student type
---------------------------------------------------------
Explore the data for each student type by doing three queries that filter for each of the student types represented in the table.

There is 67 Domestic and 201 International students in the dataset, 18 students don't give their information. So we gonna investigate what there 18 students could be. 

In [5]:
SELECT *
FROM students
WHERE inter_dom NOT LIKE 'D%' AND inter_dom NOT LIKE 'I%';

inter_dom  region  gender  ...  alone_bi  others_bi  internet_bi
0         NaN     NaN     NaN  ...       NaN        NaN          NaN
1         NaN     NaN     NaN  ...       NaN        NaN          NaN
2         NaN     NaN     NaN  ...       NaN        NaN          NaN
3         NaN     NaN     NaN  ...       NaN        NaN          NaN
4         NaN     NaN     NaN  ...       NaN        NaN          NaN
5         NaN     NaN     NaN  ...       NaN        NaN          NaN
6         NaN     NaN     NaN  ...      65.0       21.0         45.0
7         NaN     NaN     NaN  ...     203.0      247.0        223.0
8         NaN     NaN     NaN  ...       NaN        NaN          NaN
9         NaN     NaN     NaN  ...      65.0       21.0         45.0
10        NaN     NaN     NaN  ...     203.0      247.0        223.0
11        NaN     NaN     NaN  ...       NaN        NaN          NaN
12        NaN     NaN     NaN  ...       NaN        NaN          NaN
13        NaN     NaN     NaN  ...       NaN        NaN          NaN
14        NaN     NaN     NaN  ...       NaN        NaN          NaN
15        NaN     NaN     NaN  ...       NaN        NaN          NaN
16        NaN     NaN     NaN  ...       NaN        NaN          NaN
17        NaN     NaN     NaN  ...       NaN        NaN          NaN

[18 rows x 50 columns]

There is 18 null records in the data set

In [8]:
SELECT 
region, COUNT(inter_dom) AS count_inter_dom
FROM students
WHERE inter_dom = 'Inter'
GROUP BY region
ORDER BY count_inter_dom DESC;

region  count_inter_dom
0     SEA              122
1      EA               48
2      SA               18
3  Others               11
4     JAP                2

By investigate futhermore in students demographic, we know that lots of the students come from South East Asia, some come from East Asia and South Asia. 

Task 3: Find out the summary statistics of the diagnostic tests for all students
----------------------------------------------------------------
Find the summary statistics for each diagnostic test using the aggregrate functions for minimum, maximum, and average. Round the averages to two decimal places and use aliases to keep the output clean. Your aliases should be min_phq, max_phq, and avg_phq; repeat this structure for the other two tests.

Find the minimum, maximum and average for each diagnostic test (PHQ-9, SCS, ASISS).

In [1]:
SELECT 
    MIN(todep) AS min_phq,
    MAX(todep) AS max_phq,
    ROUND(AVG(todep),2) AS avg_phq,
    
    MIN(tosc) AS min_scs,
    MAX(tosc) AS max_scs,
    ROUND(AVG(tosc),2) AS avg_scs,
    
    MIN(toas) AS min_ASISS,
    MAX(toas) AS max_ASISS,
    ROUND(AVG(toas),2) AS avg_ASISS
FROM students;

min_phq  max_phq  avg_phq  min_scs  ...  avg_scs  min_asiss  max_asiss  avg_asiss
0        0       25     8.19        8  ...    37.47         36        145      72.38

[1 rows x 9 columns]

Task 4: Summarize the data for international students only
-------------------------------------------------------------------------

Find the summary statistic on domestic students and international students

In [5]:
SELECT 
    inter_dom,

    MIN(todep) AS min_phq,
    MAX(todep) AS max_phq,
    ROUND(AVG(todep),2) AS avg_phq,
    
    MIN(tosc) AS min_scs,
    MAX(tosc) AS max_scs,
    ROUND(AVG(tosc),2) AS avg_scs,
    
    MIN(toas) AS min_ASISS,
    MAX(toas) AS max_ASISS,
    ROUND(AVG(toas),2) AS avg_ASISS
FROM students
WHERE inter_dom IN ('Inter','Dom')
GROUP BY inter_dom;

inter_dom  min_phq  max_phq  ...  min_asiss  max_asiss  avg_asiss
0     Inter        0       25  ...         36        145      75.56
1       Dom        0       23  ...         36        112      62.84

[2 rows x 10 columns]

How the Age of international student impact on the score?

In [7]:
SELECT 
Age,
ROUND(AVG(todep),2) AS avg_phq,
ROUND(AVG(tosc),2) AS avg_scs,
ROUND(AVG(toas),2) AS avg_ASISS
FROM students
WHERE inter_dom = 'Inter'
GROUP BY Age
ORDER BY Age;

age  avg_phq  avg_scs  avg_asiss
0    17     4.67    37.33      70.67
1    18     8.75    34.11      80.61
2    19     8.44    37.90      74.10
3    20     7.35    38.21      73.26
4    21     9.23    37.74      75.23
5    22     8.36    38.14      70.43
6    23     9.67    32.00      81.25
7    24     4.67    42.33      74.33
8    25     6.11    37.33      80.78
9    27    10.00    35.00      42.00
10   28     3.33    39.00      71.00
11   29     3.75    43.00      63.50
12   30     9.33    41.00      97.33
13   31     5.75    43.50      80.25

How the stay length of international student impact on the score?

In [10]:
SELECT stay, 
       ROUND(AVG(todep), 2) AS average_phq, 
       ROUND(AVG(tosc), 2) AS average_scs, 
       ROUND(AVG(toas), 2) AS average_as
FROM students
WHERE inter_dom = 'Inter'
GROUP BY stay
ORDER BY stay DESC;

stay  average_phq  average_scs  average_as
0    10        13.00        32.00       50.00
1     8        10.00        44.00       65.00
2     7         4.00        48.00       45.00
3     6         6.00        38.00       58.67
4     5         0.00        34.00       91.00
5     4         8.57        33.93       87.71
6     3         9.09        37.13       78.00
7     2         8.28        37.08       77.67
8     1         7.48        38.11       72.80